packages/react-components/src/components/VideoGallery/HorizontalGallery.tsx

```diff
@@ -0,0 +1,208 @@
+   * @defaultValue `true`
+   */
+  showMuteIndicator?: boolean;
+  /** Space to leave on the left of this gallery in pixels. */
+  leftGutter?: number;
+  /** Space to leave on the right of this gallery in pixels. */
+  rightGutter?: number;
+  /** Default `false`. If set to true, video tiles will not render remote video stream  */
+  hideRemoteVideoStream?: boolean;
+}
+
+/**
+ * Renders a horizontal gallery of video tiles.
+ * @param props - HorizontalGalleryProps {@link @azure/communication-react#HorizontalGalleryProps}
+ * @returns
+ */
+export const HorizontalGallery = (props: HorizontalGalleryProps): JSX.Element => {
+  const {
+    participants,
+    onCreateRemoteStreamView,
+    onDisposeRemoteStreamView,
+    onRenderRemoteVideoTile,
+    remoteVideoViewOption,
+    onRenderAvatar,
+    showMuteIndicator,
+    hideRemoteVideoStream,
+    leftGutter = 8,
+    rightGutter = 8
+  } = props;
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const isSmallScreen = useIsSmallScreen(containerRef);
+  const containerWidth = useContainerWidth(containerRef);
+  const [page, setPage] = useState(0);
+  const [maxTiles, setMaxTiles] = useState(0);
+  const [tileHeight, setTileHeight] = useState(120);
+  const [tileWidth, setTileWidth] = useState(160);
+
+  useEffect(() => {
+    setPage(0);
+    if (isSmallScreen) {
+      setTileHeight(88);
+      setTileWidth(88);
+      const maxTiles = calculateMaxNumberOfTiles({
+        width: containerWidth - (leftGutter + rightGutter),
+        tileWidth: 88,
+        buttonsWidth: 0
+      });
+      setMaxTiles(maxTiles);
+    } else {
+      setTileHeight(120);
+      setTileWidth(160);
+      const maxTiles = calculateMaxNumberOfTiles({
+        width: containerWidth - (leftGutter + rightGutter),
+        tileWidth: 160
+      });
+      setMaxTiles(maxTiles);
+    }
+  }, [containerWidth, isSmallScreen, leftGutter, rightGutter]);
+
+  const tileSizeStyle = useMemo(
+    () => ({
+      minHeight: `${tileHeight}px`,
+      minWidth: `${tileWidth}px`,
+      maxHeight: `${tileHeight}px`,
+      maxWidth: `${tileWidth}px`
+    }),
+    [tileWidth, tileHeight]
+  );
+
+  const pageButtonSizeStyle = useMemo(
+    () => ({
+      minWidth: '1.75rem',
+      minHeight: `${tileHeight}px`,
+      maxWidth: '1.75rem',
+      maxHeight: `${tileHeight}px`
+    }),
+    [tileHeight]
+  );
+
+  const maxPages = useMemo(() => {
+    return Math.ceil(participants.length / maxTiles) - 1;
+  }, [maxTiles, participants.length]);
+
+  const changePage = (page: number): void => {
+    if (page < 0) {
+      setPage(0);
+    } else if (page >= maxPages) {
+      setPage(maxPages);
+    } else {
+      setPage(page);
+    }
+  };
+
+  const defaultOnRenderParticipants = useMemo(() => {
+    // If user provided a custom onRender function return that function.
+    if (onRenderRemoteVideoTile) {
+      return participants?.map((participant) => onRenderRemoteVideoTile(participant));
```

Comment:
Could be wrong, but my assumption to `onRenderRemoteVideoTile` would be that is only overrides the rendering of the video tile and hence would still make use of the `participants?.slice` logic? (as it should still have pages of participants)